tensorflow/lite/micro/examples/hello_world/evaluate_test.cc

```diff
@@ -127,12 +128,130 @@ int LoadFloatModelAndPerformInference() {
+  }
+
+  // This pulls in all the operation implementations we need
+  tflite::AllOpsResolver resolver;
+
+  constexpr int kTensorArenaSize = 2056;
+  uint8_t tensor_arena[kTensorArenaSize];
+
+  // Build an interpreter to run the model with
+  tflite::MicroInterpreter interpreter(model, resolver, tensor_arena,
+                                       kTensorArenaSize);
+
+  // Allocate memory from the tensor_arena for the model's tensors
+  if (interpreter.AllocateTensors() != kTfLiteOk) {
+    MicroPrintf("Allocate tensor failed.");
+    return kTfLiteError;
+  }
+
+  // Obtain a pointer to the model's input tensor
+  TfLiteTensor* input = interpreter.input(0);
+
+  // Make sure the input has the properties we expect
+  if (input == nullptr) {
+    MicroPrintf("Input tensor in null.");
+    return kTfLiteError;
+  }
+
+  // Get the input quantization parameters
+  float input_scale = input->params.scale;
+  int input_zero_point = input->params.zero_point;
+
+  // Quantize the input from floating-point to integer
+  int8_t x_quantized = x / input_scale + input_zero_point;
+  // Place the quantized input in the model's input tensor
+  input->data.int8[0] = x_quantized;
+  // Run the model and check that it succeeds
+  TfLiteStatus invoke_status = interpreter.Invoke();
+  if (invoke_status != kTfLiteOk) {
+    MicroPrintf("Interpreter invocation failed.");
+    return kTfLiteError;
+  }
+
+  // Obtain a pointer to the output tensor.
+  TfLiteTensor* output = interpreter.output(0);
+
+  // Get the output quantization parameters
+  float output_scale = output->params.scale;
+  int output_zero_point = output->params.zero_point;
+
+  // Obtain the quantized output from model's output tensor
+  int8_t y_pred_quantized = output->data.int8[0];
+  // Dequantize the output from integer to floating-point
+  float y_pred = (y_pred_quantized - output_zero_point) * output_scale;
+
+  // Check if the output is within a small range of the expected output
+  float epsilon = 0.05f;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  // Run inference on several more values and confirm the expected outputs
+  x = 1.f;
+  y_true = sin(x);
+  input->data.int8[0] = x / input_scale + input_zero_point;
+  interpreter.Invoke();
+  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
+
+  x = 3.f;
+  y_true = sin(x);
+  input->data.int8[0] = x / input_scale + input_zero_point;
+  interpreter.Invoke();
+  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
+  if (abs(y_true - y_pred) > epsilon) {
+    MicroPrintf(
+        "Difference between predicted and actual y value "
+        "is significant.");
+    return kTfLiteError;
+  }
```

Comment:
we should be running all these checks in a loop instead of repeating the code.

something like:
```cc
constexpr int kNumTestValues = 4;
float golden_inputs[kNumTestValues] = {0.f, 1.f, 3.f, 5.f};

for (int i=0; i<kNumTestValues; ++i) {
  input->data.int8[0] = golden_inputs[i] / input_scale + input_zero_point;
  interpreter.Invoke();
  y_pred = (output->data.int8[0] - output_zero_point) * output_scale;
  if (abs(sin(golden_inputs[i]) - y_pred) > epsilon) {
    MicroPrintf(
        "Difference between predicted and actual y value "
        "is significant.");
    return kTfLiteError;
  }
}
```

Let's make the same change to the float test as well.

Reply:
Done. Thank you for the suggestion.